nixos/lib/make-options-doc/default.nix

```diff
@@ -88,8 +88,49 @@ let
   # Convert the list of options into an XML file.
   optionsXML = builtins.toFile "options.xml" (builtins.toXML optionsList);
 
+  optionsNix = builtins.listToAttrs (map (o: { name = o.name; value = removeAttrs o ["name" "visible" "internal"]; }) optionsList);
+
+  # TODO: declarations: link to github
+  singleAsciiDoc = name: value: ''
+    == ${name}
+
+    ${value.description}
```

Comment:
unfortunately, options docs assume Docbook in many places. Things like `</para><para>` and `<literal>` are pretty common in descriptions. So it would be more complicated than this.

Reply:
We'll need to process those, but this is a good start for options outside of nixos (I don't expect asciidoc to be used in nixos).

Reply:
okay, so it was not clear that this is not a start if asciidocization, but just a helper function for modules written outside of Nixpkgs. And nobody should really try to use this against default Nixos modules.

And really, only `optionsNix` has to be exposed here. Everything else can be built on top of outside of Nixpkgs. This asciidoc variant is more of an example, but a useful example.

Reply:
It is a start, but it's only really useful outside of nixpkgs and partially inside nixpkgs.

> Everything else can be built on top of outside of Nixpkgs.

Nixpkgs is supposed to be the central Nix packaging repo and since Hydra doesn't allow IFD, there's little choice here. Flakes will make it easier, but as of today in order to distribute reusable Nix code, nixpkgs is the only sane place.

Think of it this way: if we never challenge docbook tooling, we'll be stuck with it forever. Some might like this, but why kill the efforts of those that don't?